modules/ROOT/pages/custom-ingress-configuration.adoc

```diff
@@ -1,77 +1,360 @@
+. A Mule application developer uses Runtime Manager to configure an application for deployment by selecting an available host, an optional subdomain, and a path combination, as specified in the ingress template. 
+. Runtime Fabric receives the application deployment request and uses the ingress configurations from the template to create a corresponding ingress object in the cluster.
 
-[IMPORTANT]
+== Example Ingress Resource Templates
+
+The following examples illustrate how to modify a Kubernetes ingress spec to create a Runtime Fabric ingress resource template. 
+
+Use the API version appropriate to your Kubernetes version:
+
+* Kubernetes v1.19 or later: 
+** networking.k8s.io/v1 
+** networking.k8s.io/v1beta1
++
+As of v1.22, the `extensions/v1beta1` and `networking.k8s.io/v1beta1` API versions of Ingress are no longer available.
+* Kubernetes v1.14 through v1.18: 
+** only networking.k8s.io/v1beta1  
+
+=== networking.k8s.io/v1
+
+[NOTE]
 ====
-On Google Kubernetes Engine (GKE), the ingress controller included with GKE provisions a separate HTTP load balancer per application by default. Before using the ingress controller provided by GKE, learn more about its behavior, exploring workarounds, or use another ingress controller if this behavior is undesirable. See link:https://help.mulesoft.com/s/article/Default-Ingress-Controller-Behavior-with-Runtime-Fabric-on-GKE[KB article] for more details.
+This template is a simplified version of the configuration. The final configuration of the template can vary depending on the controller and ingress routing rules you use. Review the documentation for your ingress controller carefully.
 ====
 
-== Procedure
+[source,yaml]
+--
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example-rtf-ingress
+  namespace: rtf #<1>                           
+  labels:
+    business-group: business-group-id
+    environment: environment-id
+spec:
+  ingressClassName: rtf-nginx #<2> 
+  tls: #<4>
+  - hosts:
+      - app-name.example.com
+    secretName: example-tls
+  rules:
+  - host: app-name.example.com #<3>
+    http:
+      paths: #<5>
+      - pathType: Prefix
+        path: /
+        backend: #<6>
+          service:
+            name: service-name
+            port:
+              name: service-port
+--
 
-. Copy the following into a text file:
+Note the following about this example:
+  
+<1>  The template must be placed in the `rtf` namespace.
+
+<2>  ingressClassName must be prefixed with `rtf-`, for example, `rtf-nginx`.
 +
+This is how Runtime Fabric recognizes the object as a template. A template that uses the `rtf-` prefix in the `ingressClassName`, for example, `rtf-nginx`, is consumed by the Runtime Fabric agent only and not by the actual ingress controller. Ingress controllers discover only resources with an `ingressClassName` value that uses the vendor-specific name, for example, `nginx` or `haproxy`.
+
+<3> TLS is optional.
```

Comment:
It will be worth mentioning that:
If the TLS section is added with a "secretName" its the responsibility of the user to make sure that the referred tls secret is created in the RTF namespace.

Reply:
```suggestion
<3> TLS is optional.
+
If the TLS section is added with a `secretName` its the responsibility of the user to make sure that the referred TLS secret is created in the RTF namespace.
```